app/api/api_v1/endpoints/data.py

```diff
@@ -17,54 +18,158 @@
+            usable_column_names[col] = col_pos
+        col_pos = col_pos + 1
+
+    #A
+    # Refuse to accept dataset if there is only date+time information.
+    # If but one of these is missing, then we have date+something or time+something which is again, useless.
+    if len(usable_column_names.items()) < 3:
+        raise HTTPException(
+            status_code=400,
+            detail="Error, .csv contains two or less usable columns, please upload a file with more columns"
+        )
+
+    new_dataset = crud.dataset.create(db=db, obj_in=BaseDataset(name=csv_file.filename))
+
+    rows = []
     for row in csv_reader:
-        if first_column:
-            # Check against existing units, if new ones add them to db.
-            for unit in row:
-                parts = unit.split(" ")
-                # Filter the first 4 cols
-                if len(parts) == 1:
-                    continue
-                # This gets the wind direction out (because it has no unit)
-                elif parts[-1].find(")") == -1:
-                    continue
-                unit_name = "".join(e + "_" for e in (parts[:len(parts)-1])).lower().strip("_")
-                unit_db = crud.unit.get_by_name(db=db, name=unit_name)
-                if unit_db:
-                    continue
-                # Check whether unit is already in db
-                symbol_name = parts[-1].strip("()")
-
-                crud.unit.create(db=db, obj_in=UnitCreate(name=unit_name, symbol=symbol_name))
-
-            first_column = False
-            continue
-
-
-        obj_in = CreateData(
-            date=datetime.datetime.strptime(row[0], "%Y-%m-%d"),
-            time=datetime.datetime.strptime(row[1], "%H:%M:%S").time(),
-            nuts3=row[2],
-            nuts2=row[3],
-            temperature_air=float(row[4].replace(",", ".")) if row[4] != "" else None,
-            relative_humidity=float(row[5].replace(",", ".")) if row[5] != "" else None,
-            precipitation=float(row[6].replace(",", ".")) if row[6] != "" else None,
-            wind_speed=float(row[7].replace(",", ".")) if row[7] != "" else None,
-            wind_direction=float(row[8].replace(",", ".")) if row[8] != "" else None,
-            wind_gust=float(row[9].replace(",", ".")) if row[9] != "" else None,
-            atmospheric_pressure=float(row[10].replace(",", ".")) if row[10] != "" else None,
-            relative_humidity_canopy=float(row[11].replace(",", ".")) if row[11] != "" else None,
-            temperature_canopy=float(row[12].replace(",", ".")) if row[12] != "" else None,
-            solar_irradiance_copernicus=float(row[13].replace(",", ".")) if row[13] != "" else None
+        try:
+            leaf_wetness = float(row[usable_column_names["leaf_wetness"]].replace(",", ".")) if "leaf_wetness" in usable_column_names else None
+
+            if leaf_wetness and (leaf_wetness < 0.0 or leaf_wetness > 1.0):
+                raise HTTPException(
```

Comment:
maybe it makes more sense to. throw a `ValueError` here or sth else but not `HTTPException`

Reply:
Alright, that makes sense